lib/ViewModels/SettingsPanelViewModel.js

```diff
@@ -22,6 +22,15 @@ var SettingsPanelViewModel = function(options) {
     this.baseMaps = defaultValue(options.baseMaps, []).slice();
     this.mouseOverBaseMap = undefined;
 
+    var that = this;
+    if (defined(options.baseMapName)) {
+        this.baseMaps.map (function(item) {
+            if (options.baseMapName === item.catalogItem.name) {
+                that.terria.baseMap = item.catalogItem;
```

Comment:
I started with it in index.js, but then it's only available it you build that in a certain way.  This is the first chance to have the basemaps list to check against.
________________________________
From: Kevin Ring [notifications@github.com]
Sent: Tuesday, 19 May 2015 10:30 AM
To: TerriaJS/terriajs
Cc: Keith Grochow
Subject: Re: [terriajs] Share maps panel settings (#698)


In lib/ViewModels/SettingsPanelViewModel.js<https://github.com/TerriaJS/terriajs/pull/698#discussion_r30560058>:

> @@ -22,6 +22,15 @@ var SettingsPanelViewModel = function(options) {
>      this.baseMaps = defaultValue(options.baseMaps, []).slice();
>      this.mouseOverBaseMap = undefined;
>
> +    var that = this;
> +    if (defined(options.baseMapName)) {
> +        this.baseMaps.map (function(item) {
> +            if (options.baseMapName === item.catalogItem.name) {
> +                that.terria.baseMap = item.catalogItem;


This needs to go somewhere else, because the base map specification needs to work even in an application that doesn't use the SettingsPanelViewModel. I'll see if I can find a better place.

—
Reply to this email directly or view it on GitHub<https://github.com/TerriaJS/terriajs/pull/698/files#r30560058>.

________________________________

The information in this e-mail may be confidential and subject to legal professional privilege and/or copyright. National ICT Australia Limited accepts no liability for any damage caused by this email or its attachments.